src/site/stages/build/process-cms-exports/tests/transformer-helpers.unit.spec.js

```diff
@@ -307,4 +310,58 @@ describe('CMS export transformer helpers', () => {
       });
     });
   });
+
+  describe('findMatchingEntities', () => {
+    const contentDir = path.join(__dirname, 'helper-test-entities');
+
+    it('should reject non-string baseTypes', () => {
+      expect(() => findMatchingEntities(contentDir, 123)).to.throw();
+    });
+
+    it('should reject paths to a non-existent directory', () => {
+      // Path to nothing
+      expect(() =>
+        findMatchingEntities(
+          path.join(__dirname, 'this-directory-does-not-exist'),
+          'node',
+        ),
+      ).to.throw();
+      // Path to a file
+      expect(() => findMatchingEntities('node', __filename)).to.throw();
+    });
+
+    it('should reject a truthy non-string subType', () => {
+      expect(() =>
+        findMatchingEntities('node', contentDir, { subType: 123 }),
+      ).to.throw();
+    });
+
+    it('should reject a truthy non-function filter', () => {
+      expect(() =>
+        findMatchingEntities('node', contentDir, { filter: 123 }),
+      ).to.throw();
+    });
+
+    it('should return all (and only) entities of a baseType', () => {
+      expect(findMatchingEntities('node', contentDir)).to.have.length(3);
+    });
+
+    it('should return only entities of a baseType and subType', () => {
+      expect(
+        findMatchingEntities('node', contentDir, { subType: 'some_type' }),
+      ).to.have.length(2);
+    });
+
+    it('should filter out entities not passing the filter function', () => {
+      expect(
+        findMatchingEntities('node', contentDir, {
+          subType: 'some_type',
+          filter: e => e.field_keep_me,
+        }),
+      ).to.have.length(1);
+    });
+
+    // Not sure how to write this test without a lot of extra wiring
+    // it('should transform the entities', () => {});
```

Comment:
:thinking: 

And I don't think it would be easy to mock/spy the `assembleEntityTrees` either.

Reply:
I _could_ pass `assembleEntityTree` as an optional parameter just for testing. :thinking: I just don't really want to, if I'm honest. Because I don't want folks to think they have to build their own `assembleEntityTree` in the transformer, nor pass it into this function if we threaded it through from `assembleEntityTree` -> `transformEntity` -> `transform` -> `findMatchingEntities`.

Reply:
Another "simpler" option might be having a `./tests/transformers/` directory that has a dummy transformer function that we can spy on and see if it gets called.